doozer/doozerlib/cli/scan_osh.py

```diff
@@ -1,29 +1,302 @@
+        for brew_package_name in packages:
+            data = packages[brew_package_name]
+
+            distgit_name = self.get_distgit_name_from_brew_nvr(data["nvr"])
+
+            image_meta: ImageMetadata = self.runtime.image_map[distgit_name]
+
+            try:
+                if not image_meta.scanning.jira_integration.enabled:
+                    self.runtime.logger.info(f"Skipping OCPBUGS creation for package {data['package_name']} "
+                                             f"since disabled in image metadata")
+                    continue
+            except AttributeError:
+                # Since this will fail until we set it up in image metadata
+                self.runtime.logger.info(f"Skipping OCPBUGS creation for package {data['package_name']} "
+                                         f"since disabled in image metadata")
+                continue
+
+            # Returns project and component name
+            _, potential_component = image_meta.get_jira_info()
+
+            # Check if ticket exits
+            summary = f"{self.version} SAST scan issues for {brew_package_name}"
+            query = f"project={self.jira_project} AND ( summary ~ '{summary}' ) AND " \
+                    "statusCategory in ('New', 'Assigned')"
+            description = "{*}{color:#FF0000}WARNING{color}{*}\n" + MESSAGE.format(package_name=brew_package_name,
+                                                                                   scan_id=data["latest_coverity_scan"],
+                                                                                   nvr=data["nvr"],
+                                                                                   brew_build_id=data["brew_build_id"]
+                                                                                   )
+
+            @retry(reraise=True, stop=stop_after_attempt(10), wait=wait_fixed(3))
+            def search_issues(query):
+                return self.jira_client.search_issues(query)
+
+            open_issues = search_issues(query)
+            self.runtime.logger.info(f"Issues found with query '{query}': {open_issues}")
+
+            fields = {
+                "project": {"key": f"{self.jira_project}"},
+                "issuetype": {"name": "Bug"},
+                "versions": [{"name": f"{self.version}.z"}],  # Affects Version/s
+                "components": [{"name": potential_component}],
+                "security": {"id": "11697"},  # Restrict to Red Hat Employee
+                "summary": summary,
+                "description": description
+            }
+
+            if len(open_issues) == 0:
+                # No tickets exist in ('New', 'Assigned') so we could consider opening a new one
+                # Do not reraise a ticket if the diff has not changed, but we would still need to update the ticket
+                # with the new build and scan
+                if not self.skip_diff_check and not self.is_latest_scan_different(
```

Comment:
Making sure I understand this - we will not open a jira ticket until there is a new defect since this scan starts running? Why not open ticket for existing ones? Are we factoring in that the dev team may have closed the previous ticket?

Reply:
So the policy that Justin suggested is that if the developers close the ticket, we will consider that as acknowledgement that they fixed the issue. So we need to raise only if we find new issues.

Reply:
I've added the [policy document](https://docs.google.com/document/d/1GsT3tvw5qf3sOSAkTSVWnMIO6vzIuirhsVhhsbK_qf0/edit#heading=h.uilnzpc6uuc7) to the PR description.

Reply:
This is currently under discussion with other Staff Engineers and managers, since the problem appears to be that we need to make sure that we don't spam devs with tickets.

Reply:
Makes sense, I'm thinking of the case of opening tickets for the first time - if an image that does not receive frequent updates, will not have a ticket created until some change which includes a new defect gets pushed, which could be months in the worst case? Maybe we can explicitly search for closed issues to get around this case?

Reply:
Or we can run it with the `--skip-diff-check` flag for the first time for all images?

Reply:
Ahh the first scan case, another thing I wrote the code for but got wiped.

Reply:
So we just need to say that its different right? Like this https://github.com/openshift-eng/art-tools/pull/141/commits/3442602132283a1ad22a78455e4217182553ba6d

Reply:
Ahh wait, I see what you're saying. So the scenario is that the first scan had vulnerabilities, we raised the ticket, devs closed it. But they didn't fix it. The second commit to that repo is after 2 months. No new vulnerabilties are found, we will not reraise the ticket as per our current policy.

Reply:
Well I guess right now, the policy is lenient so that we can ease in the devs to this new process. Once everyone gets adjusted to the new normal we can make it more strict?

> Maybe we can explicitly search for closed issues to get around this case?
 There is the `--skip-diff-check` flag as well, so we could use that for this case?